cot/src/form/fields.rs

```diff
@@ -763,232 +767,819 @@ macro_rules! impl_float_as_form_field {
+///
+/// # Example
+///
+/// ```
+/// use chrono::{Duration, FixedOffset, NaiveDateTime};
+/// use cot::form::fields::{
+///     DateTimeWithTimezoneField, DateTimeWithTimezoneFieldOptions, Step,
+/// };
+/// use cot::form::{FormField, FormFieldOptions};
+///
+/// // Suppose we want UTC+05:30 as the “base” offset, and if there’s a DST fold,
+/// // we choose the later offset (i.e. `prefer_latest = true`).
+/// let tz_offset = FixedOffset::east_opt(5 * 3600 + 30 * 60).unwrap();
+///
+/// let options = DateTimeWithTimezoneFieldOptions {
+///     min: None,
+///     max: None,
+///     readonly: Some(false),
+///     step: Some(Step::Value(Duration::seconds(60))),
+///
+///     // Use UTC+05:30 when parsing datetime from the browser.
+///     timezone: Some(tz_offset),
+///
+///     // If a given local time is ambiguous (DST fall‐back), pick the later of the two possibilities.
+///     prefer_latest: Some(true),
+/// };
+///
+/// let field = DateTimeWithTimezoneField::with_options(
+///     FormFieldOptions {
+///         id: "dt".into(),
+///         name: "dt".into(),
+///         required: true,
+///     },
+///     options,
+/// );
+/// ```
+#[derive(Debug, Default, Clone, Copy)]
+pub struct DateTimeWithTimezoneFieldOptions {
+    /// The maximum allowed datetime (with offset) for this field.
+    ///
+    /// When present, sets the `max` attribute on `<input
+    /// type="datetime-local">`, and also enforces `datetime > max` at
+    /// validation time.
+    pub max: Option<DateTime<FixedOffset>>,
+
+    /// The minimum allowed datetime (with offset) for this field.
+    ///
+    /// When present, sets the `min` attribute on `<input
+    /// type="datetime-local">`, and also enforces `datetime < min` at
+    /// validation time.
+    pub min: Option<DateTime<FixedOffset>>,
+
+    /// Whether the field should be read‐only.
+    ///
+    /// When `Some(true)`, we render `readonly="true"` on the HTML tag,
+    /// preventing user edits. When `None` or `Some(false)`, the input is
+    /// editable.
+    pub readonly: Option<bool>,
+
+    /// The increment (in seconds) between valid datetimes.
+    ///
+    /// Corresponds to the `step` attribute on `<input type="datetime-local">`.
+    /// If `None`, the browser’s default (60 seconds) is used. To override,
+    /// supply `Step::Value(Duration)` where `Duration` indicates the number
+    /// of seconds per “tick.”
+    ///
+    /// [`step` attribute]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#step
+    pub step: Option<Step<Duration>>,
+
+    /// The timezone offset to use when parsing the browser‐supplied string.
+    ///
+    /// Browsers send a naive “YYYY-MM-DDThh:mm” (no offset). If `timezone` is
+    /// `Some(offset)`, we convert that naive time into a
+    /// `DateTime<FixedOffset>` using `offset.from_local_datetime(...)`. If
+    /// `None`, defaults to UTC.
+    pub timezone: Option<FixedOffset>,
```

Comment:
Would you rather have the `timezone` and `prefer_latest` options as system-wide configs than per form field?

Reply:
That would be ideal, but currently we don't have any sensible way to pass any external config when creating forms. Do you have any ideas how to implement this?

If not, I have some ideas on reworking the form APIs so that such dependency injection is possible, and also to allow using different forms for different purposes (e.g., admin panel). I'll post a proposal soon.